pkg/controllers/management/node/cleanup.go

```diff
@@ -121,8 +121,10 @@ func (m *Lifecycle) drainNode(node *v3.Node) error {
 	}
 
 	logrus.Infof("[node-cleanup] node [%s] attempting to drain, retrying up to 3 times", nodeCopy.Spec.RequestedHostname)
-	// purposefully ignoring error, if the drain fails this falls back to deleting the node as usual
-	return wait.ExponentialBackoff(backoff, func() (bool, error) {
+	// purposefully ignoring kubectl.drain error. However, if the node fails to drain after 3 attempts
+	// wait.ExponentialBackoff will still return a wait.ErrWaitTimeout error, which must also be ignored.
+	// Otherwise, the node will not actually delete and resources will be orphaned in the provider.
+	_ = wait.ExponentialBackoff(backoff, func() (bool, error) {
```

Comment:
How does this stop at 3 attempts? Is that the wait func default?

Reply:
Right above this we build a configuration for the backoff https://github.com/rancher/rancher/blob/4064f864bc9da8251560e28ad6be82832b7970be/pkg/controllers/management/node/cleanup.go#L115-L122